client/client_test.go

```diff
@@ -6199,6 +6201,202 @@ func testExportAnnotations(t *testing.T, sb integration.Sandbox) {
 	}
 }
 
+func testExportAttestations(t *testing.T, sb integration.Sandbox) {
+	requiresLinux(t)
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	p := platforms.MustParse("linux/amd64")
+	pk := platforms.Format(p)
+
+	success := []byte(`{"success": true}`)
+	successDigest := digest.SHA256.FromBytes(success)
+
+	frontend := func(ctx context.Context, c gateway.Client) (*gateway.Result, error) {
+		res := gateway.NewResult()
+
+		// build image
+		st := llb.Scratch().File(
+			llb.Mkfile("/greeting", 0600, []byte("hello world!")),
+		)
+		def, err := st.Marshal(ctx)
+		if err != nil {
+			return nil, err
+		}
+		r, err := c.Solve(ctx, gateway.SolveRequest{
+			Definition: def.ToPB(),
+		})
+		if err != nil {
+			return nil, err
+		}
+		ref, err := r.SingleRef()
+		if err != nil {
+			return nil, err
+		}
+		_, err = ref.ToState()
+		if err != nil {
+			return nil, err
+		}
+		res.AddRef(pk, ref)
+
+		// build attestations
+		st = llb.Scratch().
+			File(llb.Mkfile("/attestation.json", 0600, success)).
+			File(llb.Mkfile("/attestation2.json", 0600, []byte{}))
+		def, err = st.Marshal(ctx)
+		if err != nil {
+			return nil, err
+		}
+		r, err = c.Solve(ctx, gateway.SolveRequest{
+			Definition: def.ToPB(),
+		})
+		if err != nil {
+			return nil, err
+		}
+		refAttest, err := r.SingleRef()
+		if err != nil {
+			return nil, err
+		}
+		_, err = ref.ToState()
+		if err != nil {
+			return nil, err
+		}
+
+		expPlatforms := &exptypes.Platforms{
+			Platforms: []exptypes.Platform{{ID: pk, Platform: p}},
+		}
+		dt, err := json.Marshal(expPlatforms)
+		if err != nil {
+			return nil, err
+		}
+		res.AddMeta(exptypes.ExporterPlatformsKey, dt)
+
+		res.AddAttestation(pk, gateway.Attestation{
+			PredicateRef:  refAttest,
+			PredicatePath: "/attestation.json",
+			PredicateType: "https://example.com/attestations/v1.0",
+			Subjects: []gateway.AttestationSubject{
+				&gateway.AttestationSubjectSelf{},
+			},
+		})
+		res.AddAttestation(pk, gateway.Attestation{
+			PredicateRef:  refAttest,
+			PredicatePath: "/attestation2.json",
+			PredicateType: "https://example.com/attestations2/v1.0",
+			Subjects: []gateway.AttestationSubject{
+				&gateway.AttestationSubjectRaw{
+					Name:   "/attestation.json",
+					Digest: successDigest,
+				},
+			},
+		})
+
+		return res, nil
+	}
+
+	target := "testattestations:latest"
+
+	_, err = c.Build(sb.Context(), SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type: ExporterImage,
+				Attrs: map[string]string{
+					"name": target,
+				},
+			},
+		},
+	}, "", frontend, nil)
+	require.NoError(t, err)
+
+	ctx := namespaces.WithNamespace(sb.Context(), "buildkit")
+	cdAddress := sb.ContainerdAddress()
+	if cdAddress != "" {
```

Comment:
Ideally would push to a local registry or do OCI export, so it works on all workers. If it complicates a lot then let me know.